search.go

```diff
@@ -355,40 +363,49 @@ func (r *SearchRequest) SetSearchBefore(before []string) {
 // UnmarshalJSON deserializes a JSON representation of
 // a SearchRequest
 func (r *SearchRequest) UnmarshalJSON(input []byte) error {
-	var temp struct {
-		Q                json.RawMessage   `json:"query"`
-		Size             *int              `json:"size"`
-		From             int               `json:"from"`
-		Highlight        *HighlightRequest `json:"highlight"`
-		Fields           []string          `json:"fields"`
-		Facets           FacetsRequest     `json:"facets"`
-		Explain          bool              `json:"explain"`
-		Sort             []json.RawMessage `json:"sort"`
-		IncludeLocations bool              `json:"includeLocations"`
-		Score            string            `json:"score"`
-		SearchAfter      []string          `json:"search_after"`
-		SearchBefore     []string          `json:"search_before"`
-	}
-
-	err := json.Unmarshal(input, &temp)
-	if err != nil {
+	var (
+		temp struct {
+			Q                json.RawMessage   `json:"query"`
+			Size             *int              `json:"size"`
+			From             int               `json:"from"`
+			Highlight        *HighlightRequest `json:"highlight"`
+			Fields           []string          `json:"fields"`
+			Facets           FacetsRequest     `json:"facets"`
+			Explain          bool              `json:"explain"`
+			Sort             []json.RawMessage `json:"sort"`
+			IncludeLocations bool              `json:"includeLocations"`
+			Score            string            `json:"score"`
+			SearchAfter      []string          `json:"search_after"`
+			SearchBefore     []string          `json:"search_before"`
+		}
+		err error
+	)
+
+	if err := json.Unmarshal(input, &temp); err != nil {
```

Comment:
`err = json.Unmarshal`

Reply:
Thank you!
I understood your comment, and in order to leave the previous commit as it is, I proceeded with a forced push and modified it.